pkg/registry/file/generatednetworkpolicy_test.go

```diff
@@ -0,0 +1,1491 @@
+								To: []softwarecomposition.NetworkPolicyPeer{
+									{
+										IPBlock: &softwarecomposition.IPBlock{
+											CIDR: "172.17.0.0/16",
+										},
+									},
+								},
+							},
+							{
+								Ports: []softwarecomposition.NetworkPolicyPort{
+									{
+										Port:     pointer.Int32(80),
+										Protocol: &protocolTCP,
+									},
+								},
+								To: []softwarecomposition.NetworkPolicyPeer{
+									{
+										IPBlock: &softwarecomposition.IPBlock{
+											CIDR: "198.17.0.2/32",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				PoliciesRef: []softwarecomposition.PolicyRef{
+					{
+						IPBlock:    "172.17.0.0/16",
+						OriginalIP: "172.17.0.2",
+						DNS:        "test.com",
+						Name:       "test",
+					},
+					{
+						IPBlock:    "198.17.0.2/32",
+						OriginalIP: "198.17.0.2",
+						DNS:        "stripe.com",
+						Name:       "stripe.com",
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+
+		got, err := generateNetworkPolicy(test.networkNeighbors, test.knownServers, timeProvider)
+
+		assert.NoError(t, err)
+
+		assert.Equal(t, test.expectedNetworkPolicy, got, test.name)
+	}
+}
+
+func TestGeneratedNetworkPolicyStorage_Get(t *testing.T) {
+	type args struct {
+		key    string
+		opts   storage.GetOptions
+		objPtr runtime.Object
+	}
+	tests := []struct {
+		name          string
+		args          args
+		create        bool
+		expectedError error
+		want          runtime.Object
+	}{
+		{
+			name: "no existing objects return empty list",
+			args: args{
+				key: "/spdx.softwarecomposition.kubescape.io/generatednetworkpolicies/kubescape/toto",
+			},
+			expectedError: storage.NewKeyNotFoundError("/spdx.softwarecomposition.kubescape.io/networkneighborses/kubescape/toto", 0),
+		},
+		{
+			name: "existing object is returned",
+			args: args{
+				key:    "/spdx.softwarecomposition.kubescape.io/generatednetworkpolicies/kubescape/toto",
+				objPtr: &v1beta1.GeneratedNetworkPolicy{},
+			},
+			expectedError: nil,
+			create:        true,
+			want: &v1beta1.GeneratedNetworkPolicy{
+				TypeMeta: v1.TypeMeta{
+					Kind:       "GeneratedNetworkPolicy",
+					APIVersion: "spdx.softwarecomposition.kubescape.io/v1beta1",
+				},
+				Spec: v1beta1.NetworkPolicy{
+					Kind:       "NetworkPolicy",
+					APIVersion: "networking.k8s.io/v1",
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							"generated-by": "kubescape",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	realStorage := NewStorageImpl(afero.NewMemMapFs(), "/")
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			generatedNetworkPolicyStorage := NewGeneratedNetworkPolicyStorage(&realStorage)
+
+			if tt.create {
+				wlObj := &softwarecomposition.NetworkNeighbors{}
+				err := realStorage.Create(context.TODO(), "/spdx.softwarecomposition.kubescape.io/networkneighborses/kubescape/toto", wlObj, nil, 0)
+				assert.NoError(t, err)
+			}
```

Comment:
issue: declaring a real storage outside the range loop will cause it to save values across multiple runs

If we use the same storage for multiple `t.Run()` calls, it will save the results, which can lead to unexpected test behavior. I had to fix something like this just recently.

```suggestion
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			realStorage := NewStorageImpl(afero.NewMemMapFs(), "/")
			generatedNetworkPolicyStorage := NewGeneratedNetworkPolicyStorage(&realStorage)

			if tt.create {
				wlObj := &softwarecomposition.NetworkNeighbors{}
				err := realStorage.Create(context.TODO(), "/spdx.softwarecomposition.kubescape.io/networkneighborses/kubescape/toto", wlObj, nil, 0)
				assert.NoError(t, err)
			}
```